### PR TITLE
Expose more SFTP file types

### PR DESCRIPTION
**Pull request by basile-henry**
As a user of the `ssh2` crate it was not possible to figure out if a
file was a named pipe, a character device, a block device, or a socket.

By changing FileType to an enum we expose more information about the
file type when it's available.

I tried to keep the API as backwards compatible as I could. Not all bit
patterns of `perm` lead to a valid/known filetype, so we end up with an
`Other` variant. Another design would have been to make
`FileStat::file_type` return an `Option<FileType>` but that would have
been a breaking API change.

### CODE REVIEW

**Review by crisidev — APPROVED** [review]
LGTM!

**Comment by crisidev** [comment]
I just need to double check that the file types are 100% correct and I will merge. Once this is merged I will release a new version of both ssh2 and libssh2-sys.

**Comment by basile-henry** [comment]
The file types come from here: https://github.com/libssh2/libssh2/blob/cfe0bf64985fd6a5db3b45ffc31a2fe3b8fd9948/include/libssh2_sftp.h#L133-L147

> Once this is merged I will release a new version of both ssh2 and libssh2-sys.

Great! :smile: Although if you prefer waiting for more changes in order to batch changes into your releases, that would be fine with me.

Do you need me to add a changelog entry?

**Comment by crisidev** [comment]
Nope, this is fine, we are not keeping a changelog as of now. I have another change i need to push, i will bundle them up.

**Comment by crisidev** [comment]
Thanks a lot for the contribution

**Review by crisidev — APPROVED** [review]

**Comment by crisidev** [comment]
Sorry, I am really really late with this PR, I have been out without my laptop for a couple of months. I am merging this right now as it looks good and will create a new release bundling a bunch of changes hopefully by the end of the week.

**Comment by crisidev** [comment]
BTW thanks a lot for keeping the API backward compatible. I can see the value of returning an `Option<FileType>` but I honestly prefer a non-breaking change here.

**Pull request merged**
